signac/contrib/migration/v1_to_v2.py

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2021 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Migrate from schema version 1 to version 2.
+
+This migration involves the following changes:
+    - Removal of the signac project id
+"""
+
+from ...common.config import _get_project_config_fn, read_config_file
+
+
+def migrate_v1_to_v2(project):
+    """Migrate from schema version 1 to version 2."""
+    config = read_config_file(_get_project_config_fn(project.root_directory()))
+    config["schema_version"] = 2
+    del config["project"]
```

Comment:
Is there a benefit to deleting the `"project"` key? Extraneous keys in the config are just ignored if I recall correctly, so I'd think we could leave it without deleting user data. We still need a schema version bump to indicate the change in how the config is interpreted.

Reply:
Why leave it? Wouldn't that be confusing?

Reply:
🤷‍♂️ I don't have a strong opinion on this. We can remove that key for clarity - I just wanted to check because we wouldn't strictly _need_ to delete this key of the "user data" as a part of a migration.

Reply:
I think we should remove it. Yes, extraneous keys would be ignored, but why would you want to leave the clutter of something from a previous schema version? And in any case, part of the logic of the version 2 schema is to separate the config completely from user data (which is why we're moving signac.rc to the hidden .signac/config).

Reply:
Cool. Let's remove it.

Reply:
Ok, sorry to piss on the parade, but thinking about it again, the only reservation I have is that some users might have put crucial (provenance) metadata into the value of this key. 😟  Thoughts?

Reply:
That was my primary concern: users might have named one project `SelfAssembly-Shape1-N=4000` and another project `CoolingRuns-Shape42-N=8000`. This key might have meaning for some users.

In practice, I know of very few users that store anything here besides a vague project name that usually matches the directory name that it is stored in. An alternative would be to move this configuration information into a project document key/value.

“Project names have been removed from signac’s database schema. Do you wish to create a project document key `project_name` with the value `{project.id}` to preserve this metadata?”

Reply:
@bdice I like that idea.

Reply:
Yes I'd support an optional move to the project document. I feel strongly that we should remove it from the config to clearly separate user data from internal metadata, so offering an optional move to the project document would also be a clear indication to the user of how such data _should_ be stored by users in the future.